src/main.rs

```diff
@@ -21,7 +24,15 @@ use crate::tree::Tree;
 use crate::ui::{App, HeaderContext};
 
 // Forbid large data size to ensure TUI performance
-const MAX_DATA_SIZE: usize = 30 * 1024 * 1024;
+static MAX_DATA_SIZE: LazyLock<usize> = LazyLock::new(|| {
+    use std::str::FromStr;
+    let megabytes = std::env::var("OTREE_MAX_SIZE")
```

Comment:
I would add `file` somewhere in the constant name

Something like


```suggestion
    let megabytes = std::env::var("OTREE_MAX_FILE_SIZE")
```

Or

```suggestion
    let megabytes = std::env::var("OTREE_MAX_DATA_SIZE")
```

Because otherwise people won't necessarily understand this constant is about a file size and not a max number of lines for example